pkg/providers/vault/interfaces.go

```diff
@@ -20,3 +20,16 @@ import "net/http"
 type Caller interface {
 	Do(req *http.Request) (*http.Response, error)
 }
+
+// TokenExpiredCallback is the callback function to handle the case when the vault token has already expired
+type TokenExpiredCallback func(expiredToken string) (replacementToken string, retry bool)
```

Comment:
This doesn't need to be a public type.  Compiler will enforce value passed is of the proper type regardless of whether this type is public or private.  Unless there's a reason for this type to be explicitly referenced outside of the vault package, it should be made private.

Reply:
true on the compiler perspective but I thought the caller might probably want to know this type without looking here :-)

Reply:
I'm not suggesting you remove the type definition.  I'm suggesting you define it as `tokenExpiredCallback` so it's private.  Caller will need to look it up anyway to determine the required signature.

Reply:
ok